roles/smartgateway/tasks/deployment.yml

```diff
@@ -0,0 +1,42 @@
+---
+# tasks file for smartgateway
+- name: Start the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: apps.openshift.io/v1
+      kind: DeploymentConfig
+      metadata:
+        name: '{{ meta.name }}-smartgateway'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: '{{ size }}'
+        template:
+          metadata:
+            labels:
+              app: smart-gateway
+          spec:
+            securityContext:
+              nonroot: true
+            containers:
+            - name: smart-gateway
+              image: '{{ container_image_path }}'
+              volumeMounts:
+              - name: sg-config
+                mountPath: /config
+              args:
+                - '-config=/config/smartgateway_config.json'
+                - '-uname=$(MY_POD_NAME)'
+                - '-servicetype={{ service_type }}'
+              env:
+                - name: MY_POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      apiVersion: v1
+                      fieldPath: metadata.name
+              ports:
+              - name: "{{ service_type }}"
```

Comment:
@csibbitt does this need to match the port name on the `Service` object?

Reply:
Hadn't actually read [this](https://kubernetes.io/docs/concepts/services-networking/service/) before, and it looks like the name is just a nice convention that I could reference instead of the port number. So the answer is, _no_.

Reply:
I think it would make sense for them to match, even if not strictly required, otherwise we're just adding more names to the same thing for no reason